@here/harp-mapview/lib/MapView.ts

```diff
@@ -1936,40 +1971,43 @@ export class MapView extends THREE.EventDispatcher {
      *
      * @param target The location to look at.
      * @param distance The distance of the camera to the target in meters.
-     * @param tiltDeg The camera tilt angle in degrees (0 is vertical), curbed below 89deg.
-     * @param headingDeg The camera heading angle in degrees and clockwise (as opposed to yaw),
+     * @param tiltDeg The camera tilt angle in degrees (0 is vertical), curbed below 89deg
+     *                @default 0
+     * @param headingDeg The camera heading angle in degrees and clockwise (as opposed to yaw)
+     *                   @default 0
      * starting north.
+     * @deprecated Use lookAt version with [[LookAtParams]] object parameter.
      */
+    lookAt(target: GeoCoordLike, distance: number, tiltDeg?: number, headingDeg?: number): void;
+
+    /**
+     * Adjusts the camera to look at a given geo coordinate with tilt and heading angles.
+     * @param params LookAtParams
+     */
+    lookAt(params: LookAtParams): void;
+
     lookAt(
-        target: GeoCoordinates,
-        distance: number,
-        tiltDeg: number = 0,
-        headingDeg: number = 0
+        targetOrParams: GeoCoordLike | LookAtParams,
+        distance?: number,
+        tiltDeg?: number,
+        headingDeg?: number
     ): void {
-        const limitedTilt = Math.min(MapViewUtils.MAX_TILT_DEG, tiltDeg);
-        // MapViewUtils#setRotation uses pitch, not tilt, which is different in sphere projection.
-        // But in sphere, in the tangent space of the target of the camera, pitch = tilt. So, put
-        // the camera on the target, so the tilt can be passed to getRotation as a pitch.
-        MapViewUtils.getCameraRotationAtTarget(
-            this.projection,
-            target,
-            -headingDeg,
-            limitedTilt,
-            this.camera.quaternion
-        );
-        MapViewUtils.getCameraPositionFromTargetCoordinates(
-            target,
-            distance,
-            -headingDeg,
-            limitedTilt,
-            this.projection,
-            this.camera.position
-        );
-        this.camera.updateMatrixWorld(true);
-        // TODO: Consider forcing entire cameras update, see: [[updateCameras]]
-        this.m_targetGeoPos.copy(target);
-        this.m_targetWorldPos.copy(this.projection.projectPoint(target));
-        this.m_targetDistance = distance;
+        if (isGeoCoordinatesLike(targetOrParams)) {
+            const zoomLevel =
+                distance !== undefined
+                    ? MapViewUtils.calculateZoomLevelFromDistance(this, distance)
+                    : undefined;
+
+            const params: LookAtParams = {
+                target: targetOrParams,
+                zoomLevel,
+                tilt: tiltDeg,
```

Comment:
Minor: doc says, default `tilt` and `heading` is `0`, but if they are not present, `undefined` here will be converted to "current" `tilt`/`heading` in `lookAtImpl`.

Reply:
done